crates/worker/src/state.rs

```diff
@@ -1,14 +1,31 @@
-#[derive(Clone, Debug)]
-pub struct State {}
+use std::sync::Arc;
+use deadpool_diesel::postgres::Pool;
+use deadpool_diesel::{Manager, Runtime};
+use tracing::info;
+use irelia_adapter::repositories::postgres::job_db::JobDBRepository;
+use irelia_common::cli_args::CliArgs;
+use irelia_core::ports::job::JobPort;
+use crate::options::Options;
 
-impl Default for State {
-    fn default() -> Self {
-        Self::new()
-    }
+#[derive(Clone)]
+pub struct State {
+    pub job_port: Arc<dyn JobPort + Send + Sync>,
 }
 
 impl State {
     pub fn new() -> Self {
-        State {}
+        let options: Options = CliArgs::default_run_or_get_options(env!("APP_VERSION"));
```

Comment:
options is injected from the main, don't do like this

Reply:
.

Reply:
completed